Assets/Scripts/Models/Utility.cs

```diff
@@ -0,0 +1,784 @@
+    public string GetDescription()
+    {
+        return UnlocalizedDescription;
+    }
+
+    public IEnumerable<string> GetAdditionalInfo()
+    {
+        yield return string.Empty;
+    }
+
+    /// <summary>
+    /// Returns the description of the job linked to the utility. NOT INMPLEMENTED.
+    /// </summary>
+    /// <returns>Job description of the job linked to the utility.</returns>
+    public string GetJobDescription()
+    {
+        return string.Empty;
+    }
+
+    /// <summary>
+    /// Gets the Context Menu Actions.
+    /// </summary>
+    /// <param name="contextMenu">The context menu to check for actions.</param>
+    /// <returns>Context menu actions.</returns>
+    public IEnumerable<ContextMenuAction> GetContextMenuActions(ContextMenu contextMenu)
+    {
+        yield return new ContextMenuAction
+        {
+            Text = "Deconstruct " + Name,
+            RequireCharacterSelected = false,
+            Action = (ca, c) => Deconstruct(this)
+        };
+        if (jobs.Count > 0)
+        {
+            for (int i = 0; i < jobs.Count; i++)
+            {
+                if (!jobs[i].IsBeingWorked)
+                {
+                    yield return new ContextMenuAction
+                    {
+                        Text = "Prioritize " + Name,
+                        RequireCharacterSelected = true,
+                        Action = (ca, c) =>
+                        {
+                            c.PrioritizeJob(jobs[0]);
+                        }
+                    };
+                }
+            }
+        }
+
+        foreach (ContextMenuLuaAction contextMenuLuaAction in contextMenuLuaActions)
+        {
+            if (!contextMenuLuaAction.DevModeOnly ||
+                Settings.GetSetting("DialogBoxSettings_developerModeToggle", false))
+            {
+                yield return new ContextMenuAction
+                {
+                    Text = contextMenuLuaAction.Text,
+                    RequireCharacterSelected = contextMenuLuaAction.RequiereCharacterSelected,
+                    Action = (cma, c) => InvokeContextMenuLuaAction(contextMenuLuaAction.LuaFunction, c)
+                };
+            }
+        }
+    }
+
+    // Make a copy of the current utility.  Sub-classes should
+    // override this Clone() if a different (sub-classed) copy
+    // constructor should be run.
+    public Utility Clone()
+    {
+        return new Utility(this);
+    }
+
+    // FIXME: These functions should never be called directly,
+    // so they probably shouldn't be public functions of Utility
+    // This will be replaced by validation checks fed to use from
+    // LUA files that will be customizable for each piece of utility.
+    // For example, a door might specific that it needs two walls to
+    // connect to.
+    private bool DefaultIsValidPosition(Tile tile)
+    {
+        bool tooCloseToEdge = tile.X < MinEdgeDistance || tile.Y < MinEdgeDistance ||
```

Comment:
We should probably make this check a function in World. And later use it in Furniture too.